packages/trie/src/trie/trie.ts

```diff
@@ -505,6 +552,19 @@ export class Trie {
       const branchNode = branchNodes[0][1]
       const branchNodeKey = branchNodes[0][0]
 
+      // Special case where one needs to delete an extra node:
```

Comment:
Does the `_formatNode` method also need to be updated? Not clear on what it does but it makes use of `_deleteFromDB` so perhaps it should also consider `useNodePruning`?

Reply:
Hmm this is weird, I have no idea why tests passed, I intended to set `_deleteFromDB` to `true` if `useNodePruning` is true, but I actually messed it up and only did so when `persistRoot` was true. 

However I think that if `formatNode` is called with `remove = true`, then earlier on we already have put that node on the delete stack, since it is part of the stack if you retrieve it by `findPath`